web/ui/src/Services/AppsController.js

```diff
@@ -95,32 +95,26 @@
 
             service.model.Endpoints.forEach(endpoint => {
                 if(endpoint.VHostList){
-                    endpoint.VHostList.forEach(vHost => endPoints.push(vHost));
+                    endpoint.VHostList.forEach(vHost => {
+                        vHost.URL = utils.publicEndpointURL($scope, vHost);
```

Comment:
anything past the `model` property of a service (or instance, host, etc) should be immutable. The view is responsible for translating this data into something it can display to users, not the controller.